File: ContainsDuplicate.java
/*
Given an array of integers, find if the array contains any duplicates. Your function should return true if any value appears at least twice in the array, and it should return false if every element is distinct.
*/

class ContainsDuplicate {
    public boolean containsDuplicate(int[] nums) {
        
         HashSet<Integer> set = new HashSet<Integer>();
         for (int n : nums){
             if(set.contains(n)){
                 return true;
             }
             set.add(n);
             
         } 
        return false;
    }
}

File: ContainsDuplicate.py
'''
Given an array of integers, find if the array contains any duplicates. Your function should return true if any value appears at least twice in the array, and it should return false if every element is distinct.

'''
class Solution:
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        dict={}
        for n in nums:
            if n not in dict:
                dict[n] = 1
            else:
                dict[n] += 1
            
        for n in nums:
            if dict[n] > 1:
                return True
            
        return False